### PR TITLE
Removed node update controller

### PR DESCRIPTION
**Pull request by Karthik-K-N**
This PR contains the changes to remove node update controller as setting provider id for node will be handled by  Power VS Cloud provider

### CODE REVIEW

**Comment by Karthik-K-N** [comment]
As https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/179 is merged, This is PR is ready for review and upon that can be merged

File: pkg/client/powervs_client.go [hunk 1]
```diff
@@ -419,3 +406,13 @@ func getCustomEndPointKeys(customEndpointsMap map[string]string) []string {
...
+
+// GetRegion return the PowerVS region
+func GetRegion() string {
+	return getEnvironmentalVariableValue(regionEnvironmentalVariable)
+}
+
+// GetZone return the PowerVS zone
+func GetZone() string {
+	return getEnvironmentalVariableValue(zoneEnvironmentalVariable)
+}
```
Comment (mkumatag) [inline pkg/client/powervs_client.go:418]: Instead of set/get from env, can we get it from `powerVSClient` struct?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: pkg/client/powervs_client.go (hunk 1 above)
Comment (Karthik-K-N) [reply pkg/client/powervs_client.go]: Updated the methods to retrieve information from `powerVSClient`

File: pkg/client/client.go [hunk 2]
```diff
@@ -33,4 +33,7 @@ type Client interface {
 	GetImages() (*models.Images, error)
 	GetNetworks() (*models.Networks, error)
 	GetCloudServiceInstances() ([]bluemixmodels.ServiceInstanceV2, error)
+
+	GetZone() string
+	GetRegion() string
```
Comment (mkumatag) [inline pkg/client/client.go:38]: do we really need to add these 2 functions too? I'm just wondering what advantages we are going go get by adding them here in the interface?!

File: pkg/client/powervs_client.go [hunk 3]
```diff
@@ -164,19 +158,8 @@ func NewValidatedClient(ctrlRuntimeClient client.Client, secretName, namespace,
...
-	}
-
-	zone := resource.RegionID
-
-	err = setEnvironmentVariables(zoneEnvironmentalVariable, zone)
-	if err != nil {
-		return nil, err
-	}
+	c.region = r
+	c.zone = resource.RegionID
```
Comment (mkumatag) [inline pkg/client/powervs_client.go:162]: is this really useful adding these values in the c struct? are those values anywhere from powerVSClient?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: pkg/actuators/machine/reconciler.go [hunk 4]
```diff
@@ -211,8 +211,7 @@ func (r *Reconciler) setProviderID(instance *models.PVMInstance) error {
 		return nil
 	}
 
-	//TODO: need to figure out a proper providerID here!
-	providerID := client.FormatProviderID(*instance.PvmInstanceID)
+	providerID := client.FormatProviderID(r.powerVSClient.GetRegion(), r.powerVSClient.GetZone(), r.providerSpec.ServiceInstanceID, *instance.PvmInstanceID)
```
Comment (Karthik-K-N) [inline pkg/actuators/machine/reconciler.go:214]: ProviderID is set here

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Pull request merged**